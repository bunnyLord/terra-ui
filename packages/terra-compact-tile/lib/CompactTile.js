'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

require('./compact-tile.scss');

var _Display = require('./Display');

var _Display2 = _interopRequireDefault(_Display);

var _Comment = require('./Comment');

var _Comment2 = _interopRequireDefault(_Comment);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  className: _react.PropTypes.string,
  layout: _react.PropTypes.oneOf(['oneColumn', 'twoColumns']),
  textEmphasis: _react.PropTypes.oneOf(['default', 'left']),
  isTruncated: _react.PropTypes.bool,
  accessoryAlignment: _react.PropTypes.oneOf(['alignTop', 'alignCenter']),
  leftAccessory: _react.PropTypes.element,
  rightAccessory: _react.PropTypes.element,
  displays: _react.PropTypes.arrayOf(_react.PropTypes.element),
  comment: _react.PropTypes.element
};

var defaultProps = {
  className: '',
  layout: 'oneColumn',
  textEmphasis: 'default',
  isTruncated: false,
  accessoryAlignment: 'alignCenter',
  leftAccessory: undefined,
  rightAccessory: undefined,
  display: [],
  comment: undefined
};

var CompactTile = function (_React$Component) {
  _inherits(CompactTile, _React$Component);

  function CompactTile() {
    _classCallCheck(this, CompactTile);

    return _possibleConstructorReturn(this, (CompactTile.__proto__ || Object.getPrototypeOf(CompactTile)).apply(this, arguments));
  }

  _createClass(CompactTile, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          className = _props.className,
          layout = _props.layout,
          textEmphasis = _props.textEmphasis,
          isTruncated = _props.isTruncated,
          accessoryAlignment = _props.accessoryAlignment,
          leftAccessory = _props.leftAccessory,
          rightAccessory = _props.rightAccessory,
          displays = _props.displays,
          comment = _props.comment,
          customProps = _objectWithoutProperties(_props, ['className', 'layout', 'textEmphasis', 'isTruncated', 'accessoryAlignment', 'leftAccessory', 'rightAccessory', 'displays', 'comment']);

      var tileClassNames = (0, _classnames2.default)(['terra-CompactTile', { 'terra-CompactTile--isTruncated': isTruncated }, _defineProperty({}, 'terra-CompactTile--' + layout, layout), _defineProperty({}, 'terra-CompactTile-accessory--' + accessoryAlignment, accessoryAlignment), className]);

      return _react2.default.createElement(
        'div',
        _extends({}, customProps, { className: tileClassNames }),
        CompactTile.renderAccessory(leftAccessory),
        _react2.default.createElement(
          'div',
          { className: 'terra-CompactTile-body' },
          CompactTile.renderRows(displays, layout, textEmphasis),
          comment
        ),
        CompactTile.renderAccessory(rightAccessory)
      );
    }
  }], [{
    key: 'renderAccessory',
    value: function renderAccessory(accessory) {
      return _react2.default.createElement(
        'div',
        { className: 'terra-CompactTile-accessory' },
        accessory
      );
    }
  }, {
    key: 'renderRows',
    value: function renderRows(displays, layout, emphasis) {
      if (displays === null || displays === undefined || !displays.length) {
        return undefined;
      }

      var displaysSlice = displays.slice(0, 7);
      var sliceValue = 1;
      if (layout === 'twoColumns') {
        sliceValue = 2;
      }

      var displayGroups = [];
      var displayGroup = [];
      var sliceLength = displaysSlice.length;
      for (var i = 0; i < sliceLength; i += 1) {
        displayGroup.push(displaysSlice[i]);

        if ((i + 1) % sliceValue === 0) {
          displayGroups.push(displayGroup);
          displayGroup = [];
        }
      }

      return _react2.default.createElement(
        'div',
        { className: 'terra-CompactTile-rowContainer' },
        displayGroups.map(function (group, index) {
          var row = CompactTile.renderRow(group, index, displayGroups.length, emphasis);
          return row;
        })
      );
    }
  }, {
    key: 'renderRow',
    value: function renderRow(row, rowIndex, rowCount, emphasis) {
      var rowKey = rowIndex;
      return _react2.default.createElement(
        'div',
        { className: 'terra-CompactTile-row', key: rowKey },
        row.map(function (display, contentIndex) {
          var contentKey = contentIndex;
          var contentClasses = CompactTile.classesForContent(rowIndex, rowCount, contentIndex, emphasis);
          return _react2.default.createElement(
            'div',
            _extends({ className: contentClasses }, { key: contentKey }),
            display
          );
        })
      );
    }
  }, {
    key: 'classesForContent',
    value: function classesForContent(rowIndex, rowCount, contentIndex, emphasis) {
      var classes = void 0;
      if (emphasis === 'left') {
        classes = CompactTile.leftEmphasisContentClassesFromIndexes(rowIndex, rowCount, contentIndex);
      } else {
        classes = CompactTile.defaultEmphasisContentClassesFromIndexes(rowIndex, rowCount);
      }
      return ['terra-CompactTile-content'].concat(classes).join(' ');
    }
  }, {
    key: 'defaultEmphasisContentClassesFromIndexes',
    value: function defaultEmphasisContentClassesFromIndexes(rowIndex, rowCount) {
      var contentSize = 'terra-CompactTile-content--primarySize';
      var contentColor = 'terra-CompactTile-content--primaryColor';

      if (rowIndex > 0) {
        contentSize = 'terra-CompactTile-content--secondarySize';
      }

      if (rowCount === 2 && rowIndex === 1) {
        contentColor = 'terra-CompactTile-content--secondaryColor';
      }

      if (rowIndex >= 2) {
        contentColor = 'terra-CompactTile-content--secondaryColor';
      }

      return [contentSize, contentColor];
    }
  }, {
    key: 'leftEmphasisContentClassesFromIndexes',
    value: function leftEmphasisContentClassesFromIndexes(rowIndex, rowCount, contentIndex) {
      if (contentIndex === 1) {
        return ['terra-CompactTile-content--secondarySize', 'terra-CompactTile-content--secondaryColor'];
      }

      return CompactTile.defaultEmphasisContentClassesFromIndexes(rowIndex, rowCount);
    }
  }]);

  return CompactTile;
}(_react2.default.Component);

CompactTile.propTypes = propTypes;
CompactTile.defaultProps = defaultProps;
CompactTile.Display = _Display2.default;
CompactTile.Comment = _Comment2.default;

exports.default = CompactTile;