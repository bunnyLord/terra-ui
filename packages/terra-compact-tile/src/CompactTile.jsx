import React, { PropTypes } from 'react';
import classNames from 'classnames';
import './compact-tile.scss';

// import Display from './Display';
// import Comment from './Comment';

export default class CompactTile extends React.Component {

  static renderAccessory(accessory) {
    return (
      <div className="terra-CompactTile-accessory">
        {accessory}
      </div>
    );
  }

  static renderRows(displays, layout, theme) {
    if (displays === null || displays === undefined || !displays.length) {
      return undefined;
    }

    const displaysSlice = displays.slice(0, 7);
    let sliceValue = 1;
    if (layout === 'twoColumns') {
      sliceValue = 2;
    }

    const displayGroups = [];
    let displayGroup = [];
    const sliceLength = displaysSlice.length;
    for (let i = 0; i < sliceLength; i += 1) {
      displayGroup.push(displaysSlice[i]);

      if ((i + 1) % sliceValue === 0) {
        displayGroups.push(displayGroup);
        displayGroup = [];
      }
    }

    return (
      <div className="terra-CompactTile-rowContainer">
        {displayGroups.map((group, index) => {
          const row = CompactTile.renderRow(group, index, displayGroups.length, theme);
          return row;
        })}
      </div>
    );
  }

  static renderRow(row, rowIndex, rowCount, theme) {
    const rowKey = rowIndex;
    return (
      <div className="terra-CompactTile-row" key={rowKey}>
        {row.map((display, contentIndex) => {
          const contentKey = contentIndex;
          const contentClasses = CompactTile.classesForContent(rowIndex,
                                                               rowCount,
                                                               contentIndex,
                                                               theme);
          return (
            <div {...{ className: contentClasses }} key={contentKey}>
              {display}
            </div>
          );
        })}
      </div>
    );
  }

  static classesForContent(rowIndex, rowCount, contentIndex, theme) {
    let classes;
    if (theme === 'leftEmphasisTheme') {
      classes = CompactTile.leftThemeContentClassesFromIndexes(rowIndex, rowCount, contentIndex);
    } else {
      classes = CompactTile.defaultThemeContentClassesFromIndexes(rowIndex, rowCount);
    }
    return ['terra-CompactTile-content'].concat(classes).join(' ');
  }

  static defaultThemeContentClassesFromIndexes(rowIndex, rowCount) {
    let contentSize = 'terra-CompactTile-content--primarySize';
    let contentColor = 'terra-CompactTile-content--primaryColor';

    if (rowIndex > 0) {
      contentSize = 'terra-CompactTile-content--secondarySize';
    }

    if (rowCount === 2 && rowIndex === 1) {
      contentColor = 'terra-CompactTile-content--secondaryColor';
    }

    if (rowIndex >= 2) {
      contentColor = 'terra-CompactTile-content--secondaryColor';
    }

    return [contentSize, contentColor];
  }

  static leftThemeContentClassesFromIndexes(rowIndex, rowCount, contentIndex) {
    if (contentIndex === 1) {
      return ['terra-CompactTile-content--secondarySize', 'terra-CompactTile-content--secondaryColor'];
    }

    return CompactTile.defaultThemeContentClassesFromIndexes(rowIndex, rowCount);
  }

  static classesForTileFromProps(layout, theme, alignment, truncated) {
    return classNames(['terra-CompactTile',
      { 'terra-CompactTile--isTruncated': truncated },
      { [`terra-CompactTile--${layout}`]: layout },
      { [`terra-CompactTile--${theme}`]: theme },
      { [`terra-CompactTile-accessory--${alignment}`]: alignment },
    ]);
  }

  render() {
    const tileClasses = CompactTile.classesForTileFromProps(this.props.layout,
                                                            this.props.theme,
                                                            this.props.accessoryAlignment,
                                                            this.props.isTruncated);

    const tileAttributes = Object.assign({}, this.props.attributes);
    tileAttributes.className = classNames([
      tileClasses,
      tileAttributes.className,
    ]);

    return (
      <div {...tileAttributes}>
        {CompactTile.renderAccessory(this.props.leftAccessory)}
        <div className="terra-CompactTile-body">
          {CompactTile.renderRows(this.props.displays, this.props.layout, this.props.theme)}
          {this.props.comment}
        </div>
        {CompactTile.renderAccessory(this.props.rightAccessory)}
      </div>
    );
  }
}

CompactTile.defaultProps = {
  layout: 'oneColumn',
  theme: 'defaultTheme',
  isTruncated: false,
  accessoryAlignment: 'alignCenter',
};

CompactTile.propTypes = {
  layout: PropTypes.oneOf(['oneColumn', 'twoColumns']),
  theme: PropTypes.oneOf(['defaultTheme', 'leftEmphasisTheme']),
  isTruncated: PropTypes.bool,
  accessoryAlignment: PropTypes.oneOf(['alignTop', 'alignCenter']),
  leftAccessory: PropTypes.element,
  rightAccessory: PropTypes.element,
  displays: PropTypes.arrayOf(PropTypes.element),
  comment: PropTypes.element,
  attributes: PropTypes.oneOfType([PropTypes.object]),
};
