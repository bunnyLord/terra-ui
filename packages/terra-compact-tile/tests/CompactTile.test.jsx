import React from 'react';
import initStoryshots from 'storyshots';
import CompactTile from '../src/CompactTile';

// Run snapshot tests for react-storybook
initStoryshots();

// Snapshot Tests
it('should render a default component', () => {
  const compactTile = shallow(<CompactTile />);
  expect(compactTile).toMatchSnapshot();
});

it('should render with 1 display', () => {
  const display1 = <Display text="display 1" />;
  const displays = [display1];
  const compactTile = shallow(<CompactTile displays={displays} />);
  expect(compactTile).toMatchSnapshot();
});

it('should render with 2 displays', () => {
  const display1 = <Display text="display 1" />;
  const display2 = <Display text="display 2" />;
  const displays = [display1, display2];
  const compactTile = shallow(<CompactTile displays={displays} />);
  expect(compactTile).toMatchSnapshot();
});

it('should render with 3 displays', () => {
  const display1 = <Display text="display 1" />;
  const display2 = <Display text="display 2" />;
  const display3 = <Display text="display 3" />;
  const displays = [display1, display2, display3];
  const compactTile = shallow(<CompactTile displays={displays} />);
  expect(compactTile).toMatchSnapshot();
});

it('should render with a display and graphic', () => {
  const testElement = <img alt="Graphic" />;
const display1 = <Display text="display 1" icon={ testElement } />;
const displays = [display1];
const compactTile = shallow(<CompactTile displays={displays} />);
expect(compactTile).toMatchSnapshot();
});

it('should render truncated display', () => {
  const display1 = <Display text="display1display1display1display1display1display1display1display1"
isTruncated={true} />;
const displays = [display1];
const compactTile = shallow(<CompactTile displays={displays} />);
expect(compactTile).toMatchSnapshot();
});

it('should render a comment', () => {
  const comment = <Comment text="comment" style="attention" isTruncated={false} />;
  const params = {
    layout: 'oneColumn',
    theme: 'defaultTheme',
    displays: [],
    comment,
  };
  const compactTile = shallow(<CompactTile {...params} />);
  expect(compactTile).toMatchSnapshot();
});

it('should render 1 left theme display', () => {
  const display1 = <Display text="display 1" />;
  const displays = [display1];
  const compactTile = shallow(<CompactTile theme="leftEmphasisTheme" displays={displays} />);
  expect(compactTile).toMatchSnapshot();
});

it('should render 2 left theme displays', () => {
  const display1 = <Display text="display 1" />;
  const display2 = <Display text="display 2" />;
  const displays = [display1, display2];
  const compactTile = shallow(<CompactTile theme="leftEmphasisTheme" displays={displays} />);
  expect(compactTile).toMatchSnapshot();
});

it('should render 3 left theme displays', () => {
  const display1 = <Display text="display 1" />;
  const display2 = <Display text="display 2" />;
  const display3 = <Display text="display 3" />;
  const displays = [display1, display2, display3];
  const compactTile = shallow(<CompactTile theme="leftEmphasisTheme" displays={displays} />);
  expect(compactTile).toMatchSnapshot();
});

it('should render a left accessory', () => {
  const testElement = <img alt="Graphic" />;
  const params = {
    layout: 'oneColumn',
    theme: 'defaultTheme',
    displays: [],
    leftAccessory: testElement,
  };
  const compactTile = shallow(<CompactTile {...params} />);
  expect(compactTile).toMatchSnapshot();
});

it('should render a right accessory', () => {
  const testElement = <img alt="Graphic" />;
  const params = {
    layout: 'oneColumn',
    theme: 'defaultTheme',
    displays: [],
    rightAccessory: testElement,
  };
  const compactTile = shallow(<CompactTile {...params} />);
  expect(compactTile).toMatchSnapshot();
});

it('should render an accessory top aligned', () => {
  const testElement = <img alt="Graphic" />;
  const params = {
    layout: 'oneColumn',
    theme: 'defaultTheme',
    displays: [],
    leftAccessory: testElement,
    accessoryAlignment: 'alignTop',
  };
  const compactTile = shallow(<CompactTile {...params} />);
  expect(compactTile).toMatchSnapshot();
});

it('should render an accessory center aligned', () => {
  const testElement = <img alt="Graphic" />;
  const params = {
    layout: 'oneColumn',
    theme: 'defaultTheme',
    displays: [],
    leftAccessory: testElement,
    accessoryAlignment: 'alignCenter',
  };
  const compactTile = shallow(<CompactTile {...params} />);
  expect(compactTile).toMatchSnapshot();
});

it('should render one column', () => {
  const display1 = <Display text="display 1" />;
  const display2 = <Display text="display 2" />;
  const displays = [display1, display2];
  const compactTile = shallow(<CompactTile layout="oneColumn" displays={displays} />);
  expect(compactTile).toMatchSnapshot();
});

it('should render two columns', () => {
  const display1 = <Display text="display 1" />;
  const display2 = <Display text="display 2" />;
  const displays = [display1, display2];
  const compactTile = shallow(<CompactTile layout="twoColumns" displays={displays} />);
  expect(compactTile).toMatchSnapshot();
});


// Prop Tests
it('should have the class terra-CompactTile--default', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper.prop('className')).toContain('terra-CompactTile terra-CompactTile--default');
});

it('should have the class terra-CompactTile--primary', () => {
  const wrapper = shallow(primaryVariety);
  expect(wrapper.prop('className')).toContain('terra-CompactTile terra-CompactTile--primary');
});


// Event Tests
it('should toggle the class u-selected on default', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-CompactTile--default').simulate('click');
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-CompactTile--default').simulate('click');
  expect(wrapper).toMatchSnapshot();
});

it('should toggle the class u-selected on primary', () => {
  const wrapper = shallow(primaryVariety);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-CompactTile--primary').simulate('click');
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-CompactTile--primary').simulate('click');
  expect(wrapper).toMatchSnapshot();
});
