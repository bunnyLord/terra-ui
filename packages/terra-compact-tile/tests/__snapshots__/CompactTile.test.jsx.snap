exports[`test should render 1 left theme display 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--oneColumn terra-CompactTile-accessory--alignCenter"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory" />
  <div
    className="terra-CompactTile-body">
    <div
      className="terra-CompactTile-rowContainer">
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--primarySize terra-CompactTile-content--primaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 1" />
        </div>
      </div>
    </div>
  </div>
  <div
    className="terra-CompactTile-accessory" />
</div>
`;

exports[`test should render 2 left theme displays 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--oneColumn terra-CompactTile-accessory--alignCenter"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory" />
  <div
    className="terra-CompactTile-body">
    <div
      className="terra-CompactTile-rowContainer">
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--primarySize terra-CompactTile-content--primaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 1" />
        </div>
      </div>
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--secondarySize terra-CompactTile-content--secondaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 2" />
        </div>
      </div>
    </div>
  </div>
  <div
    className="terra-CompactTile-accessory" />
</div>
`;

exports[`test should render 3 left theme displays 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--oneColumn terra-CompactTile-accessory--alignCenter"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory" />
  <div
    className="terra-CompactTile-body">
    <div
      className="terra-CompactTile-rowContainer">
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--primarySize terra-CompactTile-content--primaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 1" />
        </div>
      </div>
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--secondarySize terra-CompactTile-content--primaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 2" />
        </div>
      </div>
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--secondarySize terra-CompactTile-content--secondaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 3" />
        </div>
      </div>
    </div>
  </div>
  <div
    className="terra-CompactTile-accessory" />
</div>
`;

exports[`test should render a comment 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--oneColumn terra-CompactTile-accessory--alignCenter"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory" />
  <div
    className="terra-CompactTile-body">
    <Comment
      className=""
      isTruncated={false}
      text="comment"
      textStyle="attention" />
  </div>
  <div
    className="terra-CompactTile-accessory" />
</div>
`;

exports[`test should render a default component 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--oneColumn terra-CompactTile-accessory--alignCenter"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory" />
  <div
    className="terra-CompactTile-body" />
  <div
    className="terra-CompactTile-accessory" />
</div>
`;

exports[`test should render a left accessory 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--oneColumn terra-CompactTile-accessory--alignCenter"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory">
    <img
      alt="Graphic" />
  </div>
  <div
    className="terra-CompactTile-body" />
  <div
    className="terra-CompactTile-accessory" />
</div>
`;

exports[`test should render a right accessory 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--oneColumn terra-CompactTile-accessory--alignCenter"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory" />
  <div
    className="terra-CompactTile-body" />
  <div
    className="terra-CompactTile-accessory">
    <img
      alt="Graphic" />
  </div>
</div>
`;

exports[`test should render an accessory center aligned 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--oneColumn terra-CompactTile-accessory--alignCenter"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory">
    <img
      alt="Graphic" />
  </div>
  <div
    className="terra-CompactTile-body" />
  <div
    className="terra-CompactTile-accessory" />
</div>
`;

exports[`test should render an accessory top aligned 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--oneColumn terra-CompactTile-accessory--alignTop"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory">
    <img
      alt="Graphic" />
  </div>
  <div
    className="terra-CompactTile-body" />
  <div
    className="terra-CompactTile-accessory" />
</div>
`;

exports[`test should render one column 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--oneColumn terra-CompactTile-accessory--alignCenter"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory" />
  <div
    className="terra-CompactTile-body">
    <div
      className="terra-CompactTile-rowContainer">
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--primarySize terra-CompactTile-content--primaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 1" />
        </div>
      </div>
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--secondarySize terra-CompactTile-content--secondaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 2" />
        </div>
      </div>
    </div>
  </div>
  <div
    className="terra-CompactTile-accessory" />
</div>
`;

exports[`test should render truncated display 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--oneColumn terra-CompactTile-accessory--alignCenter"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory" />
  <div
    className="terra-CompactTile-body">
    <div
      className="terra-CompactTile-rowContainer">
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--primarySize terra-CompactTile-content--primaryColor">
          <Display
            className=""
            isTruncated={true}
            text="display1display1display1display1display1display1display1display1" />
        </div>
      </div>
    </div>
  </div>
  <div
    className="terra-CompactTile-accessory" />
</div>
`;

exports[`test should render two columns 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--twoColumns terra-CompactTile-accessory--alignCenter"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory" />
  <div
    className="terra-CompactTile-body">
    <div
      className="terra-CompactTile-rowContainer">
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--primarySize terra-CompactTile-content--primaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 1" />
        </div>
        <div
          className="terra-CompactTile-content terra-CompactTile-content--primarySize terra-CompactTile-content--primaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 2" />
        </div>
      </div>
    </div>
  </div>
  <div
    className="terra-CompactTile-accessory" />
</div>
`;

exports[`test should render with 1 display 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--oneColumn terra-CompactTile-accessory--alignCenter"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory" />
  <div
    className="terra-CompactTile-body">
    <div
      className="terra-CompactTile-rowContainer">
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--primarySize terra-CompactTile-content--primaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 1" />
        </div>
      </div>
    </div>
  </div>
  <div
    className="terra-CompactTile-accessory" />
</div>
`;

exports[`test should render with 2 displays 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--oneColumn terra-CompactTile-accessory--alignCenter"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory" />
  <div
    className="terra-CompactTile-body">
    <div
      className="terra-CompactTile-rowContainer">
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--primarySize terra-CompactTile-content--primaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 1" />
        </div>
      </div>
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--secondarySize terra-CompactTile-content--secondaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 2" />
        </div>
      </div>
    </div>
  </div>
  <div
    className="terra-CompactTile-accessory" />
</div>
`;

exports[`test should render with 3 displays 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--oneColumn terra-CompactTile-accessory--alignCenter"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory" />
  <div
    className="terra-CompactTile-body">
    <div
      className="terra-CompactTile-rowContainer">
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--primarySize terra-CompactTile-content--primaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 1" />
        </div>
      </div>
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--secondarySize terra-CompactTile-content--primaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 2" />
        </div>
      </div>
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--secondarySize terra-CompactTile-content--secondaryColor">
          <Display
            className=""
            isTruncated={false}
            text="display 3" />
        </div>
      </div>
    </div>
  </div>
  <div
    className="terra-CompactTile-accessory" />
</div>
`;

exports[`test should render with a display and graphic 1`] = `
<div
  className="terra-CompactTile terra-CompactTile--oneColumn terra-CompactTile-accessory--alignCenter"
  display={Array []}>
  <div
    className="terra-CompactTile-accessory" />
  <div
    className="terra-CompactTile-body">
    <div
      className="terra-CompactTile-rowContainer">
      <div
        className="terra-CompactTile-row">
        <div
          className="terra-CompactTile-content terra-CompactTile-content--primarySize terra-CompactTile-content--primaryColor">
          <Display
            className=""
            icon={
              <img
                alt="Graphic" />
            }
            isTruncated={false}
            text="display 1" />
        </div>
      </div>
    </div>
  </div>
  <div
    className="terra-CompactTile-accessory" />
</div>
`;
